contracts/oracles/CurveOracle.sol

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.11;
+pragma abicoder v1;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "../interfaces/IOracle.sol";
+import "../libraries/Sqrt.sol";
+
+import "../interfaces/ICurveRegistry.sol";
+import "../interfaces/ICurveProvider.sol";
+import "../interfaces/ICurveSwap.sol";
+
+contract CurveOracle is IOracle {
+    using Sqrt for uint256;
+
+    ICurveProvider public immutable addressProvider;
+    IERC20 private constant _NONE = IERC20(0xFFfFfFffFFfffFFfFFfFFFFFffFFFffffFfFFFfF);
+
+    constructor(ICurveProvider _addressProvider) {
+        addressProvider = _addressProvider;
+    }
+
+    function getRate(IERC20 _srcToken, IERC20 _dstToken, IERC20 connector) external view override returns (uint256 rate, uint256 weight) {
+
+        require(connector == _NONE, "connector should be NONE");
+
+        address registryAddress = addressProvider.get_address(0);
+        ICurveRegistry registry = ICurveRegistry(registryAddress);
+
+        address pool;
+        pool = registry.find_pool_for_coins(address(_srcToken), address(_dstToken));
+        require(pool != address(0), "Pool does not exist");
+
+        int128 a;
+        int128 b;
+        bool underlying;
+        (a, b, underlying) = registry.get_coin_indices(pool, address(_srcToken), address(_dstToken));
```

Comment:
Looks like you should be using `underlying`

Reply:
Sure. They are essentially the same but it can be made clearer, see my addition https://github.com/1inch/offchain-oracle/pull/26/commits/75beb5e27598c613bdb34773bea2a3da352fff7a

Reply:
This should work since it's calling the token contract and not the pool. no? Tests seem to indicate successful operation.

Reply:
Yeah, I mean that in case of underlying token pool does not actually hold them so calling balanceOf on underlying token will return 0. For example compound pool https://etherscan.io/address/0xA2B47E3D5c44877cca798226B7B8118F9BFb7A56 only stores cDAI and cUSDC in it so when you call `DAI.balanceOf(pool)` or `USDC.balanceOf(pool)` it'll return 0.

Reply:
thanks that clears it up some more. I think this (https://github.com/1inch/offchain-oracle/pull/26/commits/7ab27fe60927fa4bab64190cf56f6d137de089e4) addition should do the job